src/prettytable/prettytable.py

```diff
@@ -2273,10 +2317,16 @@ def _get_formatted_html_string(self, options):
             for field in self._field_names:
                 if options["fields"] and field not in options["fields"]:
                     continue
-                lines.append(
-                    '            <th style="padding-left: %dem; padding-right: %dem; text-align: center">%s</th>'  # noqa: E501
-                    % (lpad, rpad, escape(field).replace("\n", linebreak))
-                )
+                if options["escape_header"]:
+                    lines.append(
+                        '            <th style="padding-left: %dem; padding-right: %dem; text-align: center">%s</th>'  # noqa: E501
+                        % (lpad, rpad, escape(field).replace("\n", linebreak))
+                    )
+                else:
+                    lines.append(
+                        '            <th style="padding-left: %dem; padding-right: %dem; text-align: center">%s</th>'  # noqa: E501
+                        % (lpad, rpad, field.replace("\n", linebreak))
+                    )
```

Comment:
```suggestion

                if options["escape_header"]:
                    field = escape(field)

                lines.append(
                    '            <th style="padding-left: %dem; padding-right: %dem; text-align: center">%s</th>'  # noqa: E501
                    % (lpad, rpad, field.replace("\n", linebreak))
                )
```
And same about tests here.